Example/URBNSwiftCarousel/SourceViewController.swift

```diff
@@ -0,0 +1,158 @@
+        
+        /**
+        * The destinationViewController must be set as the transitioning delegate of the transitionController to trigger the transition controller's methods
+        */
+        destinationViewController.transitioningDelegate = transitionController
+        /**
+        * In order to use URBNSwiftCarousel, the modalPresentationStyle must be set to .Custom
+        */
+        destinationViewController.modalPresentationStyle = .Custom
+        
+        presentViewController(destinationViewController, animated: true , completion: nil)
+        
+        destinationViewController.dismissCallback = { [weak self] in
+            self?.dismissViewControllerAnimated(true, completion: nil)
+        }
+    }
+    
+    // MARK: URBNSwCarouselTransitioning
+    /**
+    * Required methods for the custom zoom view controller transition
+    */
+    func imageForGalleryTransition() -> UIImage {
+        guard let img = selectedCellForTransition?.imageView.image else { return UIImage() }
+        return img
+    }
+    
+    /**
+     * When your SourceViewController calls presentViewController(destinationViewController), the SOURCE view controller calls this method.
+     *
+     * When dismissViewController is called by either the source or destination, the DESTINATION view controller calls this method.
+     *
+     * Calculate and return the frame you want the animating imageView to return to.
+     */
+    func fromImageFrameForGalleryTransitionWithContainerView(containerView: UIView) -> CGRect {
+        guard let imgFrame = selectedCellForTransition?.imageView.urbn_imageFrame() else { return CGRectZero }
+        return containerView.convertRect(imgFrame, fromView: selectedCellForTransition)
+    }
+    
+    /**
+     * When your SourceViewController calls presentViewController(destinationViewController), the DESTINATION view controller calls this method.
+     *
+     * When dismissViewController is called by either the source or destination, the SOURCE view controller calls this method.
+     *
+     * Calculate and return the frame you want the animating imageView to zoom out to.
+     */
+    func toImageFrameForGalleryTransitionWithContainerView(containerView: UIView, sourceImageFrame: CGRect) -> CGRect {
+        guard let frameToReturnTo = selectedCellForTransition?.imageView.frame, selectedCell = selectedCellForTransition else { return CGRectZero }
+        let size = UIImageView.urbn_aspectFitSizeForImageSize(sourceImageFrame.size, rect: frameToReturnTo)
+        let toImageWidth = size.width
+        let toImageHeight = size.height
+        let convertedRect = containerView.convertRect(selectedCell.frame, fromView: selectedCollectionViewForTransition)
+        let originX = CGRectGetMidX(convertedRect) - size.width/2
+        let originY = CGRectGetMidY(convertedRect) - size.height/2
+        return CGRectMake(originX, originY, toImageWidth, toImageHeight)
```

Comment:
TODO possibly include a convenience method for zooming back and forth between the same images.